File: One-Time Pad (OTP) – Theoretically Unbreakable.go
package main

import (
	"crypto/rand"
	"encoding/base64"
	"fmt"
	"log"
)

// GenerateRandomKey creates a random key of the given length
func GenerateRandomKey(length int) []byte {
	key := make([]byte, length)
	_, err := rand.Read(key)
	if err != nil {
		log.Fatal(err)
	}
	return key
}

// OneTimePadEncrypt encrypts plaintext using OTP with the given key
func OneTimePadEncrypt(plainText []byte, key []byte) []byte {
	if len(plainText) != len(key) {
		log.Fatal("Key length must match plaintext length")
	}
	cipherText := make([]byte, len(plainText))
	for i := range plainText {
		cipherText[i] = plainText[i] ^ key[i] // XOR operation
	}
	return cipherText
}

// OneTimePadDecrypt decrypts OTP-encrypted text using the given key
func OneTimePadDecrypt(cipherText []byte, key []byte) []byte {
	return OneTimePadEncrypt(cipherText, key) // Decryption is the same as encryption
}

func main() {
	plainText := "Hello, Secure World!"
	plainBytes := []byte(plainText)

	// Generate a random key
	key := GenerateRandomKey(len(plainBytes))

	// Encrypt and Decrypt
	encryptedBytes := OneTimePadEncrypt(plainBytes, key)
	decryptedBytes := OneTimePadDecrypt(encryptedBytes, key)

	// Encode encrypted data in Base64 for readability
	encryptedBase64 := base64.StdEncoding.EncodeToString(encryptedBytes)
	keyBase64 := base64.StdEncoding.EncodeToString(key)

	fmt.Println("Plain Text:", plainText)
	fmt.Println("Generated Key (Base64):", keyBase64)
	fmt.Println("Encrypted Text (Base64):", encryptedBase64)
	fmt.Println("Decrypted Text:", string(decryptedBytes))
}

File: One-Time Pad (OTP) – Theoretically Unbreakable.java
import java.nio.charset.StandardCharsets;
import java.security.SecureRandom;
import java.util.Base64;

public class OneTimePad {
    
    // Generate a random key of the same length as the plaintext
    public static byte[] generateRandomKey(int length) {
        SecureRandom random = new SecureRandom();
        byte[] key = new byte[length];
        random.nextBytes(key);
        return key;
    }

    // Encrypt using One-Time Pad
    public static byte[] encrypt(byte[] plainText, byte[] key) {
        if (plainText.length != key.length) {
            throw new IllegalArgumentException("Key length must match plaintext length");
        }
        byte[] cipherText = new byte[plainText.length];
        for (int i = 0; i < plainText.length; i++) {
            cipherText[i] = (byte) (plainText[i] ^ key[i]); // XOR operation
        }
        return cipherText;
    }

    // Decrypt using One-Time Pad
    public static byte[] decrypt(byte[] cipherText, byte[] key) {
        return encrypt(cipherText, key); // Decryption is the same as encryption
    }

    public static void main(String[] args) {
        String plainText = "Hello, Secure World!";
        byte[] plainBytes = plainText.getBytes(StandardCharsets.UTF_8);
        
        byte[] key = generateRandomKey(plainBytes.length);
        byte[] encryptedBytes = encrypt(plainBytes, key);
        byte[] decryptedBytes = decrypt(encryptedBytes, key);
        
        // Encode encrypted data in Base64 to make it readable
        String encryptedBase64 = Base64.getEncoder().encodeToString(encryptedBytes);
        String decryptedText = new String(decryptedBytes, StandardCharsets.UTF_8);
        
        System.out.println("Plain Text: " + plainText);
        System.out.println("Generated Key (Base64): " + Base64.getEncoder().encodeToString(key));
        System.out.println("Encrypted Text (Base64): " + encryptedBase64);
        System.out.println("Decrypted Text: " + decryptedText);
    }
}
